packages/esm-patient-chart-app/src/deceased/deceased.resource.ts

```diff
@@ -1,134 +1,133 @@
-    deathDate: toDateObjectStrict(
-      toOmrsIsoString(new Date(dayjs(deceasedDate).year(), dayjs(deceasedDate).month(), dayjs(deceasedDate).date())),
-    ),
+  const payload: DeathPayload = {
     causeOfDeath: selectedCauseOfDeathValue,
     dead: true,
   };
-  return changePatientDeathStatus(payload.deathDate, true, personUuid, payload.causeOfDeath, abortController);
+
+  if (deceasedDate) {
+    payload.deathDate = new Date(deceasedDate.getFullYear(), deceasedDate.getMonth(), deceasedDate.getDay());
+  } else {
+    payload.deathDate = null;
+  }
+
+  return changePatientDeathStatus(personUuid, payload, abortController);
 }
 
 export function markPatientAlive(personUuid: string, abortController: AbortController) {
-  return changePatientDeathStatus(null, false, personUuid, null, abortController);
+  return changePatientDeathStatus(
+    personUuid,
+    {
+      deathDate: null,
+      causeOfDeath: null,
+      dead: false,
+    },
+    abortController,
+  );
 }
 
 export function useConceptAnswers(conceptUuid: string) {
-  const { data, error, isValidating } = useSWR<{ data: { answers: ConcentAnswersResponse } }, Error>(
+  const { data, error, isValidating } = useSWR<{ data: ConceptAnswersResponse }, Error>(
     `/ws/rest/v1/concept/${conceptUuid}`,
-    openmrsFetch,
+    (url) => (conceptUuid ? openmrsFetch(url) : undefined),
+    {
+      shouldRetryOnError(err) {
+        return err instanceof Response && err.status !== 404;
+      },
+    },
   );
-  const result = useMemo(() => {
-    return {
-      conceptAnswers: data?.data?.answers ?? null,
-      isConceptLoading: !data && !error,
-      conceptError: error,
-      isConceptAnswerValidating: isValidating,
-    };
-  }, [data, error, isValidating]);
-  return result;
-}
 
-export function usePersonfromPatient(patientUuid: string) {
-  const { data, error, isValidating, mutate } = useSWR<{ data: PersonFetchResponse }, Error>(
-    `/ws/rest/v1/patient/${patientUuid}`,
-    openmrsFetch,
-  );
-  const result = useMemo(() => {
-    return {
-      personObject: data ? data?.data?.person : null,
-      personError: error,
-      isPersonLoading: !data && !error,
-      isPersonError: error,
-      isPersonValidating: isValidating,
-      mutate,
-    };
-  }, [data, error, isValidating, mutate]);
-  return result;
+  return {
+    conceptAnswers: data?.data?.answers ?? ([] as ConceptAnswer[]),
+    isConceptLoading: !data && !error,
+    conceptError: error,
+    isConceptAnswerValidating: isValidating,
+  };
 }
 
 export function useCauseOfDeathConcept() {
   const { data, error, isValidating } = useSWR<{ data: CauseOfDeathFetchResponse }>(
     `/ws/rest/v1/systemsetting/concept.causeOfDeath`,
     openmrsFetch,
+    {
+      shouldRetryOnError(err) {
+        return err instanceof Response && err.status !== 404;
+      },
+    },
   );
   const result = useMemo(() => {
     return {
-      value: data ? data?.data.value : null,
+      value: data?.data?.value ?? undefined,
```

Comment:
Should we bubble the error up to the component?

Reply:
Yes... I the component doesn't work properly if there is no CauseOfDeath concept, but hopefully that can be done as a follow-up PR?

Reply:
Sure